sql/sql_repl.cc

```diff
@@ -2043,6 +2043,18 @@ send_event_to_slave(binlog_send_info *info, Log_event_type event_type,
     return "run 'before_send_event' hook failed";
   }
 
+
+  /*
+    We want to store data slave_info on primary that will be used for
+    show_slave_hosts_callback. We are interested in GTID sent by primary
+    to replica.
+  */
+  if (info->thd->semi_sync_slave)
+  {
+    strncpy(info->thd->slave_info->gtid_state_sent.log_file,
```

Comment:
This should probably happen between lines  2079 and 2081; so we update after we have successfully written. I also think it can be applied universally (not just when `info->thd->semi_sync_slave`).